### PR TITLE
ReLU

### PR DESCRIPTION
**Pull request by migeed-z**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #61310
* #61265
* #61239
* #61238
* #61188
* **#61150**

Differential Revision: [D29625826](https://our.internmc.facebook.com/intern/diff/D29625826)

### CODE REVIEW

File: test/fx/test_gradual_type.py [hunk 1]
```diff
@@ -406,5 +406,59 @@ def forward(self, x: TensorType((5, 2, 3, 4))):
...
+        graph = ast_rewriter.trace(B)
+        traced = GraphModule(ast_rewriter.root, graph, "gm")
+
+        tc = GraphTypeChecker({}, traced)
+        tc.type_check()
+
+        for n in traced.graph.nodes:
+            if n.target == 'output':
+                assert isinstance(n.type, TensorType)
+                assert torch.Size(n.type.__args__) == B.forward(torch.rand(2, 2, 4, 5)).size()
```
Comment (eellison) [inline test/fx/test_gradual_type.py:461]: you should know that resnet output is fixed at [1, 1000] so checking the output doesn't really check the inference well

**Comment by migeed-z** [comment]
@migeed-z has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29625826).

**Review by jamesr66a — APPROVED** [review]
image

**Comment by migeed-z** [comment]
@migeed-z has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29625826).

**Comment by migeed-z** [comment]
@migeed-z has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29625826).

**Comment by migeed-z** [comment]
@migeed-z has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29625826).

**Pull request closed without merging**